src/main/scala/viper/gobra/ast/frontend/Ast.scala

```diff
@@ -1038,6 +1037,19 @@ case class PMultisetType(elem : PType) extends PGhostLiteralType
 /** The type of option types. */
 case class POptionType(elem : PType) extends PGhostLiteralType
 
+
+case class PDomainType(funcs: Vector[PDomainFunction], axioms: Vector[PDomainAxiom]) extends PGhostLiteralType with PUnorderedScope
```

Comment:
Just to be sure, you decided not to support parametric domains to be consistent with Go's lack of support for generics right?

Reply:
We do not need generics right now and eventually there will be true ADT support